cppcache/test/LoggingTest.cpp

```diff
@@ -0,0 +1,759 @@
+    apache::geode::client::Log::close();
+
+    if (boost::filesystem::exists(testLogFileName)) {
+      boost::filesystem::remove(testLogFileName);
+    }
+
+    std::map<int32_t, boost::filesystem::path> rolledFiles;
+    LoggingTest::findRolledFiles(boost::filesystem::current_path().string(),
+                                 rolledFiles);
+    for (auto& item : rolledFiles) {
+      boost::filesystem::remove(item.second);
+    }
+  }
+
+  virtual void SetUp() { scrubTestLogFiles(); }
+
+  virtual void TearDown() { scrubTestLogFiles(); }
+
+ public:
+  static void writeRolledLogFile(const boost::filesystem::path& logdir,
+                                 int32_t rollIndex) {
+    auto rolledPath =
+        logdir / boost::filesystem::path("LoggingTest-" +
+                                         std::to_string(rollIndex) + ".log");
+    auto rolledFile = fopen(rolledPath.string().c_str(), "w");
+    fwrite("Test", 1, 4, rolledFile);
+    fclose(rolledFile);
+  }
+
+  static int numOfLinesInFile(const char* fname) {
+    char line[2048];
+    char* read;
+    int ln_cnt = 0;
+    FILE* fp = fopen(fname, "r");
+    if (fp == nullptr) {
+      return 0;
+    }
+    while (!!(read = fgets(line, sizeof line, fp))) {
+      ++ln_cnt;
+    }
+
+    if (!feof(fp)) {
+      fclose(fp);
+      return -2;
+    }
+    fclose(fp);
+    return ln_cnt;
+  }
+
+  static int expected(LogLevel level) {
+    int expected = static_cast<int>(level);
+    if (level >= LogLevel::Default) {
+      expected--;
+    }
+    return expected;
+  }
+
+  static int expectedWithBanner(LogLevel level) {
+    int expected = LoggingTest::expected(level);
+    if (level != LogLevel::None) {
+      expected += LENGTH_OF_BANNER;
+    }
+    return expected;
+  }
+
+  static void verifyLineCountAtLevel(LogLevel level) {
+    apache::geode::client::Log::init(level, testLogFileName);
+
+    apache::geode::client::Log::error("Error Message");
+    apache::geode::client::Log::warning("Warning Message");
+    apache::geode::client::Log::info("Info Message");
+    apache::geode::client::Log::config("Config Message");
+    apache::geode::client::Log::fine("Fine Message");
+    apache::geode::client::Log::finer("Finer Message");
+    apache::geode::client::Log::finest("Finest Message");
+    apache::geode::client::Log::debug("Debug Message");
+
+    int lines = LoggingTest::numOfLinesInFile(testLogFileName.c_str());
+
+    ASSERT_TRUE(lines == LoggingTest::expectedWithBanner(level));
+
+    apache::geode::client::Log::close();
+    boost::filesystem::remove(testLogFileName.c_str());
+  }
+
+  static void findRolledFiles(
+      const std::string& logFilePath,
+      std::map<int32_t, boost::filesystem::path>& rolledFiles) {
+    const auto basePath =
+        boost::filesystem::absolute(boost::filesystem::path(logFilePath)) /
+        testLogFileName;
+    const auto filterstring = basePath.stem().string() + "-(\\d+)\\.log$";
+    const boost::regex my_filter(filterstring);
```

Comment:
Why `boost::regex` and not `std::regex`?

Reply:
Uhhh, because I didn't know that was a thing and I just assumed.  I'll switch to std::, that's much better, obvs.

Reply:
Done